lib/slack.rb

```diff
@@ -1,10 +1,110 @@
 #!/usr/bin/env ruby
+require 'table_print'
+require 'httparty'
+require 'dotenv'
 
-def main
-  puts "Welcome to the Ada Slack CLI!"
+Dotenv.load
+
+require_relative 'user'
+require_relative 'channel'
+require_relative 'recipient'
+require_relative 'slack_api_error'
+require_relative 'workspace'
 
-  # TODO project
+VALID_INPUT = ["list users", "list channels", "select user", "select channel", "details", "send message", "quit"]
 
+def list_options
+  list_of_options = "\n~~MENU~~\n1. List Users\n2. List Channels\n3. Select User\n4. Select Channel\n5. Details\n6. Send message\n7. Quit\nWhich option would you like?"
+  return list_of_options
+end
+
+def handle_bad_input(user_input)
+  until VALID_INPUT.include? user_input
+    puts "Error! Your input (#{user_input}) is invalid. Please choose one of the following options and try again."
+    puts list_options
+    user_input = gets.chomp.downcase
+  end
+  return user_input
+end
+
+def id_or_name(list, user_choice)
```

Comment:
Naming these `recipients` instead of `list` would have been clearer.

```suggestion
def id_or_name(recipients, user_choice)
```